### ISSUE TITLE
Auto recompile/redeploy fails for REST/SpringBoot app

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#3113]
Steps to recreate:
* Create new app: REST/SpringBoot
* Observe successful creation of build pipeline
* Observe successful creation of Che workspace
* Edit source files - observe auto compile does not occur

![screenshot from 2018-04-12 22-19-03](https://user-images.githubusercontent.com/642621/38713615-0a1478a0-3ea0-11e8-9398-2471c0c28f4d.png)

Error seen in dev-machine window:

```
2018-04-13 02:12:21,131[s polled Thread]  [INFO ] [o.e.c.p.m.server.rmi.RmiClient 130]  - Port/Name:7487/MavenRemoteServerImpl1c91bf26 
2018-04-13 02:12:21,848[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 1023]       - Deployment of web application archive [/home/user/che/ws-agent/webapps/ROOT.war] has finished in [6,254] ms 
2018-04-13 02:12:21,851[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 632]  - Starting ProtocolHandler ["http-nio-4401"] 
2018-04-13 02:12:21,860[main]             [INFO ] [o.a.catalina.startup.Catalina 683]   - Server startup in 6453 ms 
2018/04/13 02:12:33 Start new terminal. 
2018-04-13 02:15:24,482[equestProcessor]  [ERROR] [o.e.j.internal.core.util.Util 1018]  - Status ERROR: org.eclipse.jdt.core code=0 Invalid ZIP archive: /home/user/.m2/repository/org/jboss/shrinkwrap/resolver/shrinkwrap-resolver-depchain/2.2.6/shrinkwrap-resolver-depchain-2.2.6.pom null 
2018-04-13 02:15:24,482[equestProcessor]  [ERROR] [o.e.j.internal.core.util.Util 1018]  - Status ERROR: org.eclipse.jdt.core code=0 Invalid ZIP archive: /home/user/.m2/repository/org/jboss/shrinkwrap/resolver/shrinkwrap-resolver-depchain/2.2.6/shrinkwrap-resolver-depchain-2.2.6.pom null 
2018-04-13 02:15:24,523[equestProcessor]  [ERROR] [o.e.j.internal.core.util.Util 1018]  - Status ERROR: org.eclipse.jdt.core code=0 Invalid ZIP archive: /home/user/.m2/repository/org/apache/tomcat/tomcat/8.5.20.redhat-5/tomcat-8.5.20.redhat-5.pom null 
2018-04-13 02:15:24,537[equestProcessor]  [ERROR] [o.e.j.internal.core.util.Util 1018]  - Status ERROR: org.eclipse.jdt.core code=0 Invalid ZIP archive: /home/user/.m2/repository/org/apache/tomcat/tomcat/8.5.20.redhat-5/tomcat-8.5.20.redhat-5.pom null 


```

**Comment by l0rd** [comment]
@ldimaggi I am closing this one since it looks identical to #3130. Please re-open if you consider that issues are different.

**Comment by ldimaggi** [comment]
#3130 refers to a different quickstart (HealthCheck/SpringBoot) failing

This issue refers to the REST/SpringBoot quickstart failing. 

The same solution may solve the issue for both quickstarts, but I think it's better to keep both issues open as they refer to different quickstarts. Thanks!

**Comment by l0rd** [comment]
@ldimaggi makes sense. I've set the same labels as #3130 (assigned to `team/analytics/wizard`)

**Comment by animuk** [comment]
tried with building in the Che editor for REST/Spring boot booster. for me build is successful. Below is the screenshot.
![che_build1](https://user-images.githubusercontent.com/11207106/39353835-7e35bbb8-4a26-11e8-9bb3-77add77a4db6.png)
This issue seems to happen sometime in che while it is building within . 
Someone from the team/che should check that.

**Comment by sunilk747** [comment]
Seems a error from che side. I cloned the quickstart project, build it and it's working for me. Then I made some changes to the quickstart again build it, it works for me both the time. I don't think it's a error from launcher side. Any body from che team mind looking at it.
cc @animuk

**Comment by GeorgeActon** [comment]
@l0rd See comments above

**Comment by ldimaggi** [comment]
This is a Che auto-recompile/redeploy issue. I am still seeing this issue.

Steps to recreate:
* Create new app: REST/SpringBoot
* Observe successful creation of build pipeline
* Observe successful creation of Che workspace
* Run the quickstart, access its preview endpoint
* Edit GreetingProperties.java (replace "Hello" with "Goodbye" or any other string)
* Observe auto recompile/redeploy does not occur

**Comment by l0rd** [comment]
I have done some investigation and auto-recompile never worked on Che for spring boot.

[Back in november](https://github.com/snowdrop/spring-boot-http-booster/pull/61) we added [spring boot dev tools](https://github.com/snowdrop/spring-boot-http-booster/pull/61) dependency in pom.xml for all boosters and that enabled **auto re-deploy** but only  after a **manual re-build** of the spring boot application (c.f. https://github.com/redhat-developer/rh-che/issues/254#issuecomment-335816550). This is how IntelliJ works too and at the time we considered it acceptable. Anyway it's unlikely that we will be able to fix that for GA (cc @ashumilova @vparfonov)

@ldimaggi @qodfathr do you still think label `P0` is adequate?

**Comment by ibuziuk** [comment]
@ldimaggi it is expected that auto recompile/redeploy will not occur for spring-boot since auto recompile / redeployment happens only after the re-building the project - https://github.com/redhat-developer/rh-che/issues/254#issuecomment-335816550 

So, the sequence of actions for  auto-recompile/redeploy is the following:

- execute run command
- open app url
- change java file
- execute build command
- see changes  in the app

![spring-boot-hot-deployment](https://user-images.githubusercontent.com/1461122/31444126-92e2983c-ae9b-11e7-9971-ac452797f57e.gif)

**Comment by ibuziuk** [comment]
@slemeur @bmicklea @qodfathr are you ok to remove P0 label based on comments above ?

**Comment by slemeur** [comment]
Yes this experience is the one that is provided with other tools.

@ldimaggi : Could we update the tests on this ?

**Comment by nimishamukherjee** [comment]
@animuk should we (app-creation) look into this?

**Comment by bartoszmajsak** [comment]
Based on comments from @slemeur here and in #3130 I'm closing both issues.

**Comment by bartoszmajsak** [comment]
@slemeur @ldimaggi is there a parent feature request as I see this topic being discussed in other issues such as #3130?

**Comment by slemeur** [comment]
I don't think we have a parent issue for this one.

**Comment by ldimaggi** [comment]
Closing as duplicate of - https://github.com/openshiftio/openshift.io/issues/3130 - it's the same issue.
